domains/client/block-builder/src/lib.rs

```diff
@@ -174,6 +177,18 @@ where
 
         api.initialize_block(parent_hash, &header)?;
 
+        let mut inherent_extrinsics = Self::create_inherents(parent_hash, &api, inherent_data)?;
+        extrinsics.push_front(
+            inherent_extrinsics
+                .pop_front()
+                .expect("Timestamp inherent must always exist; qed"),
+        );
+
+        // rest of the inherents, set_code if exists will go in the last
+        for inherent_extrinsic in inherent_extrinsics {
+            extrinsics.push_back(inherent_extrinsic)
+        }
```

Comment:
Nope, it needs to come in the last since `set_code` will consume all the remaning block weight. So none of the extrinsic will go through. This check is not valid here

Reply:
Well, the check can cause panic:
https://github.com/subspace/subspace/blob/5d06efec94312dfeb70aa9a10137ae56c6c2dfbe/domains/pallets/executive/src/lib.rs#L290-L292

I guess we need to handle this properly by either:
- Update/remove the check
- Change the behavior of "set_code will consume all the remaning block weight"

since both are provided by the upstream, we may need to do some research about what they are used for and whether necessary.

Reply:
I'm actually more surprised how the tests passed as I was relying on them to show me more info. I think it is fine to keep the check as is and update the set_code to not consume full block weight then

Reply:
The domain runtime upgrade test is removed during the v2 refactoring since the way of maintaining domain runtime code is changed but unfortunately we forgot to add a new test...

Reply:
🤦🏼